### PR TITLE
z_bg_goron_oyu OK

### PR DESCRIPTION
**Pull request by mzxrules**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---
Supersedes pr #206

### CODE REVIEW

File: src/overlays/actors/ovl_En_Clear_Tag/z_en_clear_tag.c [hunk 1]
```diff
@@ -776,7 +776,7 @@ void EnClearTag_DrawEffects(Actor* thisx, GlobalContext* globalCtx) {
     s16 i;
     s16 j;
     Vec3f vec;
-    WaterBox* waterBox;
+    CollisionHeader* waterBox;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Clear_Tag/z_en_clear_tag.c:779]: Emmm, what?

File: src/overlays/actors/ovl_Bg_Goron_Oyu/z_bg_goron_oyu.c [hunk 2]
```diff
@@ -8,14 +8,18 @@
 
 #define FLAGS 0x00000030
 
-#define THIS ((BgGoronOyu*)thisx)
-
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Goron_Oyu/z_bg_goron_oyu.c:12]: We haven't decided what we are going to do with the `THIS`macro, so I suggest to bring it back and use it where appropriate

File: src/overlays/actors/ovl_Bg_Goron_Oyu/z_bg_goron_oyu.c [hunk 3]
```diff
@@ -28,31 +32,162 @@ const ActorInit Bg_Goron_Oyu_InitVars = {
...
+        ActorCutscene_StartAndSetUnkLinkFields(this->initialActorCutscene, &this->dyna.actor);
+        this->actionFunc = func_80B40160;
+    } else {
+        ActorCutscene_SetIntentToPlay(this->initialActorCutscene);
+    }
+}
+
+void func_80B40160(BgGoronOyu* this, GlobalContext* globalCtx) {
+    static Vec3f D_80B40780 = { 0, 0, 0 };
+    Math_StepToF(&this->unk_164, 0.0f, 0.2f);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_Bg_Goron_Oyu/z_bg_goron_oyu.c:68]: Please add an empty line between the variables declarations and the actual code
```suggestion
    static Vec3f D_80B40780 = { 0, 0, 0 };

    Math_StepToF(&this->unk_164, 0.0f, 0.2f);
```

**Review by m4xw — APPROVED** [review]
looks generally good to me

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_Bg_Goron_Oyu/z_bg_goron_oyu.c [hunk 4]
```diff
@@ -28,31 +34,163 @@ const ActorInit Bg_Goron_Oyu_InitVars = {
...
+        if (BgCheck_EntityRaycastFloor2(globalCtx, &globalCtx->colCtx, &poly, &pos2) < this->waterBoxPos.y) {
+            pos2.y = this->waterBoxPos.y + 10.0f;
+            EffectSsIceSmoke_Spawn(globalCtx, &pos2, &vel2, &D_801D15B0, scale);
+        }
+    }
+}
+
+void BgGoronOyu_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgGoronOyu* this = THIS;
+    s32 pad;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_Bg_Goron_Oyu/z_bg_goron_oyu.c:155]:
```suggestion
    BgGoronOyu* this = THIS;
    s32 pad;
```
If this is a GlobalContext upcast pad, it tends to go above the Actor recast.

**Review by EllipticEllipsis — APPROVED** [review]
One minor thing, but I'll not let it stand in the way of approval.

File: src/overlays/actors/ovl_Bg_Goron_Oyu/z_bg_goron_oyu.h [hunk 5]
```diff
@@ -8,12 +8,15 @@ struct BgGoronOyu;
 typedef void (*BgGoronOyuActionFunc)(struct BgGoronOyu*, GlobalContext*);
 
 typedef struct BgGoronOyu {
-    /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0x18];
-    /* 0x015C */ BgGoronOyuActionFunc actionFunc;
-    /* 0x0160 */ char unk_160[0x20];
+    /* 0x000 */ DynaPolyActor dyna;
+    /* 0x15C */ BgGoronOyuActionFunc actionFunc;
+    /* 0x160 */ u8 unk_160[4];
```
Comment (Kenix3) [inline src/overlays/actors/ovl_Bg_Goron_Oyu/z_bg_goron_oyu.h:13]:
```suggestion
    /* 0x160 */ UNK_TYPE1 unk_160[4];
```

**Review by Kenix3 — CHANGES_REQUESTED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**